make examples failing

Error: Command failed: docker exec libdragon make examples
    ctest.c: In function 'main':
    ctest.c:77:27: error: too few arguments to function 'get_accessories_present'
             int accessories = get_accessories_present();
                               ^~~~~~~~~~~~~~~~~~~~~~~
    In file included from /usr/local/mips64-elf/include/libdragon.h:49:0,
                     from ctest.c:5:
    /usr/local/mips64-elf/include/controller.h:183:5: note: declared here
     int get_accessories_present(struct controller_data * data);

--- a/README.html
+++ b/README.html
@@ -1,0 +1,125 @@
+<!DOCTYPE html>
+    <html>
+    <head>
+        <meta http-equiv="Content-type" content="text/html;charset=UTF-8">
+        <title>Libdragon</title>
+        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.0/dist/katex.min.css" integrity="sha384-9eLZqc9ds8eNjO3TmqPeYcDj8n+Qfa4nuSiGYa6DjLNcv9BtN69ZIulL9+8CqC9Y" crossorigin="anonymous">
+        <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/markdown.css">
+        <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/highlight.css">
+        <link href="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.css" rel="stylesheet" type="text/css">
+        <style>
+.task-list-item { list-style-type: none; } .task-list-item-checkbox { margin-left: -20px; vertical-align: middle; }
+</style>
+        <style>
+            body {
+                font-family: -apple-system, BlinkMacSystemFont, 'Segoe WPC', 'Segoe UI', 'Ubuntu', 'Droid Sans', sans-serif;
+                font-size: 14px;
+                line-height: 1.6;
+            }
+        </style>
+        
+        <script src="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.js"></script>
+    </head>
+    <body>
+        <h1 id="libdragon">Libdragon</h1>
+<p>This is a simple library for N64 that allows one to code using the gcc compiler suite and nothing else. No proprietary library is needed.</p>
+<p>To make managing the toolcain easier, a docker container is used. Node.js is used to interact with the docker container for multi-platform support. You can inspect index.js if you prefer not to use node, but it makes things easier in general when working with docker.</p>
+<p>On a machine with node.js (&gt;= 7.6) &amp; docker you can simply do in this repository's root;</p>
+<pre><code>npm run download
+</code></pre>
+<p>to download the toolchain image from docker repository. If you instead prefer to build it on your computer do;</p>
+<pre><code>npm run init
+</code></pre>
+<p>This will build and start the container and may take a while as it will initialize the docker container and build the toolchain from scratch.</p>
+<p>Then you will be able to work on the files simultaneously with the container and any built binaries will be available in your workspace as it is mounted on the container.
+You will need to share your working drive from docker UI for it to be able to access your workspace for Windows hosts.</p>
+<p>You can also start / stop the docker container using;</p>
+<pre><code>npm start
+npm stop
+</code></pre>
+<p>To build the examples do;</p>
+<pre><code>npm run make examples
+</code></pre>
+<p>Toolchain wrapper can also run make inside the container for you with the make command;</p>
+<p>e.g to run clean all on root;</p>
+<pre><code>npm run make clean
+</code></pre>
+<p>The toolchain make command will be only run at the root-level. Use -C flag to make a directory instead;</p>
+<pre><code>npm run make -C your/path
+</code></pre>
+<p>Please note that the path should be unix-compatible, so you should not use auto completion on non-unix systems.</p>
+<p>If you export <code>N64_BYTE_SWAP</code> environment variable with a value of true (<code>export N64_BYTE_SWAP=true</code>), you can generate byte-swapped <code>.v64</code> rom files. If this is not present the Makefiles will default to not swapped <code>.z64</code> files.</p>
+<p>To use the toolchain's host make command with byte swap enabled, pass a make variable like so;</p>
+<pre><code>npm run make examples N64_BYTE_SWAP=true
+</code></pre>
+<p>You can also permanently set <code>BYTE_SWAP</code> for docker container in <code>index.js</code> and stop/start it for changes to take effect.</p>
+<p>If you need more control over the toolchain container bash into it with;</p>
+<pre><code>docker exec -i -t libdragon /bin/bash
+</code></pre>
+<p>It is also possible to install libdragon as a global NPM module, making it possible to invoke it as;</p>
+<pre><code>libdragon download
+libdragon init
+libdragon start
+libdragon stop
+libdragon make [...params]
+</code></pre>
+<p>Keep in mind that the same docker container (with the default name of libdragon) will be used for the global and git cloned libdragon instances.</p>
+<p>You can install libdragon as an NPM dependency by <code>npm install libdragon --save</code> in order to use docker in your other N64 projects. In this case, your project name will be used as the container name and this is shared among all NPM projects using that name. Again, your project's root is mounted on the docker image. Then above commands can only be used as NPM scripts in your package.json, such as;</p>
+<pre><code>&quot;scripts&quot;: {
+    &quot;init&quot;: &quot;libdragon download&quot;,
+    &quot;build&quot;: &quot;libdragon make&quot;,
+    &quot;clean&quot;: &quot;libdragon make clean&quot;
+}
+</code></pre>
+<p>Finally, you can make an NPM package that a <code>libdragon</code> project can depend on. Just include a Makefile on the repository root with recipes for <code>all</code> and <code>install</code>. On the depending project, after installing libdragon and the dependency with <code>npm install [dep name]</code>, one can install libdragon dependencies on the current docker container using package.json scripts.</p>
+<p>For example this package.json;</p>
+<pre><code>{
+    &quot;name&quot;: &quot;libdragonDependentProject&quot;,
+    &quot;version&quot;: &quot;1.0.0&quot;,
+    &quot;description&quot;: &quot;&quot;,
+    &quot;scripts&quot;: {
+        &quot;build&quot;: &quot;libdragon make&quot;,
+        &quot;clean&quot;: &quot;libdragon make clean&quot;,
+        &quot;install&quot;: &quot;libdragon install&quot;
+    },
+    &quot;dependencies&quot;: {
+        &quot;ed64&quot;: [version],
+        &quot;libdragon&quot;: [version]
+    }
+}
+</code></pre>
+<p>will download the docker image, run it with the name <code>libdragonDependentProject</code> and run <code>make &amp;&amp; make install</code> for ed64 upon running <code>npm install</code>. This is an experimental dependency management.</p>
+<h1 id="rsp-assembly">RSP assembly</h1>
+<p>Libdragon uses assembly macros to program the RSP chip defined in <code>ucode.S</code>. These mainly wrap <code>cop2</code>, <code>lwc2</code> and <code>swc2</code> instructions.</p>
+<p>The syntax is similar to that of Nintendo's but with a few changes. For example if we take <code>vabs</code> instruction;</p>
+<pre><code>vabs vd, vs, vt[e]
+</code></pre>
+<p>it becomes;</p>
+<pre><code>vabs vd, vs, vt, e
+</code></pre>
+<p>and element (<code>e</code>) is always required. It is also similar for load/store instructions. As an example, <code>sbv</code> instruction;</p>
+<pre><code>sbv vt[element], offset(base)
+</code></pre>
+<p>becomes;</p>
+<pre><code>sbv vt, element, offset, base
+</code></pre>
+<p>Basically all operands are required and separated by commas.</p>
+<p>While using these custom instructions, you should use <code>v0</code>-<code>v31</code>for naming vector registers and <code>s0</code>-<code>s31</code> for naming scalar registers.</p>
+<h1 id="original-documentation-for-reference">Original documentation for reference</h1>
+<p>To get started from scratch, follow the following steps:</p>
+<ol>
+<li>Create a directory and copy the build script there from the tools/ directory.</li>
+<li>Read the comments in the build script to see what additional packages are needed.</li>
+<li>Run ./build from the created directory, let it build and install the toolchain.</li>
+<li>Make libdragon by typing 'make' at the top level.</li>
+<li>Install libdragon by typing 'make install' at the top level.</li>
+<li>Install libpng-dev if not already installed.</li>
+<li>Make the tools by typing 'make tools' at the top level.</li>
+<li>Install the tools by typing 'make tools-install' at the top level.</li>
+<li>Compile the examples by typing 'make examples' at the top level.</li>
+</ol>
+<p>You are now ready to run the examples on your N64.
+For more information, visit <a href="http://www.dragonminded.com/n64dev/">http://www.dragonminded.com/n64dev/</a></p>
+
+    </body>
+    </html>

--- a/README.md
+++ b/README.md
@@ -60,11 +60,7 @@ It is also possible to install libdragon as a global NPM module, making it possi
 
 Keep in mind that the same docker container (with the default name of libdragon) will be used for the global and git cloned libdragon instances.
 
-You can install libdragon as an NPM dependency by `npm install libdragon --save` in order to use docker in your other N64 projects. In this case, your project name will be used as the container name and this is shared among all NPM projects using that name. Again, your project's root is mounted on the docker image. Then above commands become;
-
-    ./node_modules/.bin/libdragon download
-
-When using them in your package.json scripts, it is possible to omit `./node_modules/.bin/` such as;
+You can install libdragon as an NPM dependency by `npm install libdragon --save` in order to use docker in your other N64 projects. In this case, your project name will be used as the container name and this is shared among all NPM projects using that name. Again, your project's root is mounted on the docker image. Then above commands can only be used as NPM scripts in your package.json, such as;
 
     "scripts": {
         "init": "libdragon download",
@@ -72,11 +68,7 @@ When using them in your package.json scripts, it is possible to omit `./node_mod
         "clean": "libdragon make clean"
     }
 
-Finally, you can make an NPM package that a `libdragon` project can depend on. Just include a Makefile on the repository root with recipes for `all` and `install`. On the depending project, after installing libdragon and the dependency with `npm install [dep name]`, one can install libdragon dependencies on the current docker container by;
-
-    ./node_modules/.bin/libdragon install
-
-or using package.json scripts.
+Finally, you can make an NPM package that a `libdragon` project can depend on. Just include a Makefile on the repository root with recipes for `all` and `install`. On the depending project, after installing libdragon and the dependency with `npm install [dep name]`, one can install libdragon dependencies on the current docker container using package.json scripts.
 
 For example this package.json;
 

--- a/examples/ctest/ctest.c
+++ b/examples/ctest/ctest.c
@@ -74,7 +74,8 @@ int main(void)
         printf( "Controller 3 %spresent\n", (controllers & CONTROLLER_3_INSERTED) ? "" : "not " );
         printf( "Controller 4 %spresent\n", (controllers & CONTROLLER_4_INSERTED) ? "" : "not " );
 
-        int accessories = get_accessories_present();
+        struct controller_data output;
+        int accessories = get_accessories_present( &output );
 
         printf( "Accessory 1 %spresent %s\n", (accessories & CONTROLLER_1_INSERTED) ? "" : "not ", 
                                               (accessories & CONTROLLER_1_INSERTED) ? format_type( identify_accessory( 0 ) ) : "" );

--- a/examples/mptest/mptest.c
+++ b/examples/mptest/mptest.c
@@ -48,7 +48,8 @@ int main(void)
                     /* Read accessories present, throwing away return.  If we don't do this, then
                        initialization routines in the identify_accessory() call will fail once we
                        remove and insert a new accessory while running */
-                    get_accessories_present();
+                    struct controller_data output;
+                    get_accessories_present( &output );
 
                     /* Make sure they don't have a rumble pak inserted instead */
                     switch( identify_accessory( i ) )

--- a/examples/mputest/mputest.c
+++ b/examples/mputest/mputest.c
@@ -46,7 +46,8 @@ int main(void)
                     /* Read accessories present, throwing away return.  If we don't do this, then
                        initialization routines in the identify_accessory() call will fail once we
                        remove and insert a new accessory while running */
-                    get_accessories_present();
+                    struct controller_data output;
+                    get_accessories_present( &output );
 
                     /* Make sure they don't have a rumble pak inserted instead */
                     switch( identify_accessory( i ) )

--- a/examples/vrutest/vrutest.c
+++ b/examples/vrutest/vrutest.c
@@ -51,7 +51,8 @@ int main(void)
 
         if( controllers & CONTROLLER_1_INSERTED )
         {
-            int accessories = get_accessories_present();
+            struct controller_data output;
+            int accessories = get_accessories_present( &output );
 
             if( (accessories & CONTROLLER_4_INSERTED) && identify_accessory( 3 ) == ACCESSORY_VRU )
             {

--- a/index.js
+++ b/index.js
@@ -82,8 +82,8 @@ const availableActions = {
         return new Promise(async (resolve, reject) => {
           try {
             await runCommand('docker exec ' + options.PROJECT_NAME + ' mkdir -p /' + options.PROJECT_NAME + '/.tmp/' + name + '/');
-            await runCommand('docker cp ' + p + '/. ' + options.PROJECT_NAME + ':/' + options.PROJECT_NAME + '/.tmp/' + name + '/');
-            await runCommand('docker exec ' + options.PROJECT_NAME + '[ -f /' + options.PROJECT_NAME + '/.tmp/' + name + '/Makefile ] &&  make -C ./.tmp/' + name + '/ && make -C ./.tmp/' + name + '/ install');
+            await runCommand('docker cp "' + paths[0] + '/." ' + options.PROJECT_NAME + ':/' + options.PROJECT_NAME + '/.tmp/' + name + '/');
+            await runCommand('docker exec ' + options.PROJECT_NAME + ' /bin/bash -c "[ -f /' + options.PROJECT_NAME + '/.tmp/' + name + '/Makefile ] &&  make -C ./.tmp/' + name + '/ && make -C ./.tmp/' + name + '/ install"');
             resolve();
           } catch(e) {
             reject(e);

--- a/index.js
+++ b/index.js
@@ -81,9 +81,10 @@ const availableActions = {
         }
         return new Promise(async (resolve, reject) => {
           try {
-            await runCommand('docker exec ' + options.PROJECT_NAME + ' mkdir -p /' + options.PROJECT_NAME + '/.tmp/' + name + '/');
-            await runCommand('docker cp "' + paths[0] + '/." ' + options.PROJECT_NAME + ':/' + options.PROJECT_NAME + '/.tmp/' + name + '/');
-            await runCommand('docker exec ' + options.PROJECT_NAME + ' /bin/bash -c "[ -f /' + options.PROJECT_NAME + '/.tmp/' + name + '/Makefile ] &&  make -C ./.tmp/' + name + '/ && make -C ./.tmp/' + name + '/ install"');
+            const relativePath = path.relative(options.MOUNT_PATH, paths[0]).replace(new RegExp('\\' + path.sep), path.posix.sep);
+            const containerPath = path.posix.join('/', options.PROJECT_NAME, relativePath, '/');
+            const makePath = path.posix.join(containerPath, 'Makefile');
+            await runCommand('docker exec ' + options.PROJECT_NAME + ' /bin/bash -c "[ -f ' + makePath + ' ] &&  make -C ' + containerPath + ' && make -C ' + containerPath + ' install"');
             resolve();
           } catch(e) {
             reject(e);

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "libdragon",
-  "version": "1.1.1",
+  "version": "1.1.2",
   "description": "This is a simple library for N64 that allows one to code using the gcc compiler suite and nothing else.  No proprietary library is needed.  For more information, visit http://www.dragonminded.com/n64dev/",
   "main": "index.js",
   "engines": {

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "libdragon",
-  "version": "1.1.2",
+  "version": "1.1.3",
   "description": "This is a simple library for N64 that allows one to code using the gcc compiler suite and nothing else.  No proprietary library is needed.  For more information, visit http://www.dragonminded.com/n64dev/",
   "main": "index.js",
   "engines": {

